src/main/java/teammates/ui/template/AdminActivityLogTableRow.java

```diff
@@ -0,0 +1,160 @@
+package teammates.ui.template;
+
+import java.util.Calendar;
+import java.util.TimeZone;
+
+import teammates.common.util.ActivityLogEntry;
+import teammates.common.util.Const;
+import teammates.common.util.TimeHelper;
+import teammates.common.util.Url;
+
+public class AdminActivityLogTableRow {
+
+    private static final int TIME_TAKEN_EXPECTED = 10000;
+    private static final int TIME_TAKEN_MODERATE = 20000;
+
+    private ActivityLogEntry activityLog;
+
+    public AdminActivityLogTableRow(ActivityLogEntry entry) {
+        activityLog = entry;
+    }
+
+    // --------------- Additional generated fields ---------------
+
+    public String getUserHomeLink() {
+        switch (activityLog.getRoleWithoutMasquerade()) {
+        case Const.ActivityLog.ROLE_STUDENT:
+            return Url.addParamToUrl(Const.ActionURIs.STUDENT_HOME_PAGE,
+                                     Const.ParamsNames.USER_ID, activityLog.getGoogleId());
+        case Const.ActivityLog.ROLE_INSTRUCTOR:
+            return Url.addParamToUrl(Const.ActionURIs.INSTRUCTOR_HOME_PAGE,
+                                     Const.ParamsNames.USER_ID, activityLog.getGoogleId());
+        default:
+            return null;
+        }
+    }
+
+    public boolean getHasUserHomeLink() {
+        return activityLog.getRoleWithoutMasquerade().contains(Const.ActivityLog.ROLE_STUDENT)
+                || activityLog.getRoleWithoutMasquerade().contains(Const.ActivityLog.ROLE_INSTRUCTOR);
+    }
+
+    public String getUserIdentity() {
+        String googleId = activityLog.getGoogleId();
+        if (!googleId.contentEquals(Const.ActivityLog.AUTH_NOT_LOGIN)
+                && !googleId.contentEquals(Const.ActivityLog.UNKNOWN)) {
+            return googleId;
+        }
+
+        String email = activityLog.getEmail();
+        if (email != null && !email.contentEquals(Const.ActivityLog.UNKNOWN)) {
+            return email;
+        }
+
+        String name = activityLog.getName();
+        if (name != null && !name.contentEquals(Const.ActivityLog.UNKNOWN)) {
+            return name;
+        }
+        return "";
+    }
+
+    public boolean getHasUserEmail() {
+        return !Const.ActivityLog.UNKNOWN.contentEquals(activityLog.getEmail());
+    }
+
+    // --------------- 'is' fields to determine css class ---------------
+
+    public boolean getIsUserAdmin() {
```

Comment:
In JSP, `data.isUserAdmin` will be mapped to `data.getIsUserAdmin()` method, this method is for the usage in JSP.